### ISSUE TITLE
DOCUMENTATION: Istio AuthorizationPolicy webpage doesn't describe CUSTOM action allowed rules

### ISSUE THREAD

**Issue by cinglese** [istio/istio.io#12328]
When implementing a AuthorizationPolicy resource with a CUSTOM action, there are different validation rules that apply versus the ALLOW or DENY actions. The [current Istio documentation](https://istio.io/latest/docs/reference/config/security/authorization-policy/#AuthorizationPolicy-Action) does not describe these differences.

I suggest that the [AuthorizationPolicy.Action section -> CUSTOM subsection](https://istio.io/latest/docs/reference/config/security/authorization-policy/#AuthorizationPolicy-Action) is updated to include that the following rules are **NOT supported** with a CUSTOM action:
- Rule.from.source.namespaces
- Rule.from.source.notNamespaces
- Rule.from.source.principals
- Rule.from.source.notPrincipals
- Rule.from.source.requestPrincipals
- Rule.from.source.notRequestPrincipals
- Rule.when.key == "source.namespace"
- Rule.when.key == "source.principal"
- Rule.when.key == "request.auth.*"

Reference:  https://github.com/istio/istio/blob/fcec7a3657c09254bb83a59bb05fbe26b477eae5/pkg/config/validation/validation.go#L1943 (Lines 1943-1976)

**Comment by cinglese** [comment]
@yangminzhu

**Comment by rolfedh** [comment]
The source page for this generated doc topic, https://github.com/rolfedh/istio.io/blob/master/content/en/docs/reference/config/security/authorization-policy/index.html, indicates:
```

---
WARNING: THIS IS AN AUTO-GENERATED FILE, DO NOT EDIT. PLEASE MODIFY THE ORIGINAL SOURCE IN THE 'https://github.com/istio/api' REPO
source_repo: https://github.com/istio/api
title: Authorization Policy
description: Configuration for access control on workloads.
location: https://istio.io/docs/reference/config/security/authorization-policy.html
layout: protoc-gen-docs
generator: protoc-gen-docs
schema: istio.security.v1.AuthorizationPolicy
weight: 20
aliases: [/docs/reference/config/authorization/authorization-policy]
number_of_entries: 9
---
```

**Comment by ericvn** [comment]
@rolfedh Correct. This file is copied from https://github.com/istio/api/blob/master/security/v1/authorization_policy.pb.html#L746-L779. 

One needs to make the updates in https://github.com/istio/api/blob/master/security/v1/authorization_policy.proto, run the `make gen` there to create a new pb.html, and the automated jobs will pull that into the reference section of the docs. Note that the `master` branch typically updates preliminary.istio.io and the current release's branch will update istio.io, so the istio/api PR may need cherry-picking before it shows up on the docs pages.

**Comment by kfaseela** [comment]
@aryan16 : looks like this is completed?
